apps/web/src/components/admin/toggles/NavItemsManager.tsx

```diff
@@ -193,3 +206,90 @@ export function NavItemDialog() {
 		</Dialog>
 	);
 }
+
+interface EditNavItemDialogProps {
+	existingName: string;
+	existingUrl: string;
+	existingEnabled: boolean;
+}
+
+function EditNavItemDialog({
+	existingName,
+	existingUrl,
+	existingEnabled,
+}: EditNavItemDialogProps) {
+	const [name, setName] = useState<string>(existingName);
+	const [url, setUrl] = useState<string>(existingUrl);
+	const [open, setOpen] = useState(false);
+
+	const { execute } = useAction(editItem, {
+		onSuccess: () => {
+			console.log("Success");
+			setOpen(false);
+			toast.success("NavItem edited successfully!");
+		},
+		onError: () => {
+			toast.error("Error editing NavItem");
+		},
+	});
+
+	return (
+		<Dialog open={open} onOpenChange={setOpen}>
+			<DialogTrigger asChild>
+				<Button>Edit Item</Button>
+			</DialogTrigger>
+			<DialogContent className="sm:max-w-[425px]">
+				<DialogHeader>
+					<DialogTitle>Edit Item</DialogTitle>
+					<DialogDescription>
+						Edit an existing item shown in the non-dashboard navbar
+					</DialogDescription>
+				</DialogHeader>
+				<div className="grid gap-4 py-4">
+					<div className="grid grid-cols-4 items-center gap-4">
+						<Label htmlFor="name" className="text-right">
+							Name
+						</Label>
+						<Input
+							onChange={(e) => setName(e.target.value)}
+							id="name"
+							placeholder="A Cool Hyperlink"
+							className="col-span-3"
+							value={name}
+						/>
+					</div>
+					<div className="grid grid-cols-4 items-center gap-4">
+						<Label htmlFor="name" className="text-right">
+							URL
+						</Label>
+						<Input
+							onChange={(e) => setUrl(e.target.value)}
+							id="name"
+							placeholder="https://example.com/"
+							className="col-span-3"
+							value={url}
+						/>
+					</div>
+				</div>
+				<DialogFooter>
+					<Button
+						onClick={() => {
+							console.log("Running Action");
+							if (!name || !url)
+								return alert("Please fill out all fields.");
+
+							execute({
+								enabled: existingEnabled,
+								existingName,
+								name,
+								url,
+							});
+						}}
+					>
+						Edit
```

Comment:
Instead of "edit", can you change this to something like "done". Also, can the button have some sort of loading state attached such as a [swirl](https://lucide.dev/icons/loader-circle) or something.